odoo/addons/base/views/ir_profile_views.xml

```diff
@@ -43,6 +43,30 @@
         </field>
     </record>
 
+    <record id="enable_profiling_wizard" model="ir.ui.view">
+        <field name="name">Enable profiling</field>
+        <field name="model">base.enable.profiling.wizard</field>
+        <field name="arch" type="xml">
+            <form string="Enable profiling">
+                <div class="alert alert-warning" role="alert">
+                    <b>Profiling is currently disabled on this database.</b><br/>
+                    Profiling is a developer feature that should be used with caution on production database.<br/>
+                    Please don't activate this feature without knowing what you are doing.<br/>
+                    Enabling profiling here will allow all user to activate profiling on their session.<br/>
+                    Profiling can be disabled at any moment in the settings.<br/>
```

Comment:
Maybe this can be outside the warning? Possibly with a link to the relevant setting page?